x-pack/legacy/plugins/actions/server/lib/get_create_task_runner_function.ts

```diff
@@ -29,19 +35,46 @@ export function getCreateTaskRunnerFunction({
   encryptedSavedObjectsPlugin,
   spaceIdToNamespace,
   getBasePath,
+  isSecurityEnabled,
 }: CreateTaskRunnerFunctionOptions) {
   return ({ taskInstance }: TaskRunnerOptions) => {
     return {
       run: async () => {
-        const { spaceId, id, params } = taskInstance.params;
+        const { spaceId, actionTaskParamsId } = taskInstance.params;
         const namespace = spaceIdToNamespace(spaceId);
-        const basePath = getBasePath(spaceId);
+
+        const {
+          attributes: { actionId, params, apiKey },
+        } = await encryptedSavedObjectsPlugin.getDecryptedAsInternalUser<ActionTaskParams>(
+          'action_task_params',
+          actionTaskParamsId,
+          { namespace }
+        );
+
+        const requestHeaders: Record<string, string> = {};
+        if (isSecurityEnabled && !apiKey) {
+          throw new ExecutorError(
+            'API key is required. The attribute "apiKey" is missing.',
+            undefined,
+            false
+          );
+        } else if (isSecurityEnabled) {
+          requestHeaders.authorization = `ApiKey ${apiKey}`;
+        }
+
```

Comment:
Should we be emulating the following logic which is done in alerting's `getCreateTaskRunnerFunction`?

```
        // Ensure API key is still valid and user has access
        const {
          attributes: { alertTypeParams, actions, interval },
          references,
        } = await services.savedObjectsClient.get<RawAlert>('alert', alertId);
```

Reply:
Agreed. I have it to a more common function that is called from the `getCreateTaskRunnerFunction` here: https://github.com/elastic/kibana/pull/41389/files#diff-ad1005666e8db3977bbe696e0cbc0680R36. The reason I put it there is because we can call execute from an API endpoint just like coming from task manager (without using API keys) and both now ensure the user in context can read the action. The `execute.ts` file is the common entry point for both. It could be moved into the `routes/execute` and `getCreateTaskRunnerFunction`.